core/src/main/java/org/lflang/analyses/cast/IfNormalFormAstVisitor.java

```diff
@@ -0,0 +1,92 @@
+package org.lflang.analyses.cast;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An AST visitor that converts an original AST into the If Normal Form. See "Bounded Model Checking
+ * of Software using SMT Solvers instead of SAT Solvers" for details about the If Normal Form
+ * (https://link.springer.com/chapter/10.1007/11691617_9).
+ *
+ * <p>There are several requirements for an AST to be in INF: 1. There should be a single
+ * StatementSequence (SS) root node; 2. Each child of the SS node should be an IfBlockNode; 3.
+ * Variables in a subsequent child should be marked as "primed" versions of those in the previous
+ * child.
+ *
+ * <p>Limitations: 1. The implementation does not take the scope into account. E.g. "int i = 0; {
+ * int j = 0; }" is treated the same as "int i = 0; int j = 0;". 2. Due to the above limitation, the
+ * implementation assumes that each variable has a unique name. E.g. "{ int i = 0; }{ int i = 0; }"
+ * is an ill-formed program.
+ *
+ * <p>In this program, visit() is the normalise() in the paper.
+ */
+public class IfNormalFormAstVisitor extends CBaseAstVisitor<Void> {
+
+  public CAst.StatementSequenceNode INF = new CAst.StatementSequenceNode();
+
+  @Override
+  public Void visitStatementSequenceNode(
```

Comment:
Here, `IfNormalFormAstVisitor` builds a new AST by traversing a C AST. Since this traversal only updates the new AST variable, the visit methods do not need to return anything, so we need to set the type argument of `CBaseAstVisitor` to `void`. However, `void` is a keyword in Java, not a class, and so `Void` needs to be used here instead. `Void` is a class in Java and can be used as a type argument in generics where you don't expect to use the result type. But since it has no usable constructors and its only instance is null, methods that have Void as a return type would have to return null. I admit it looks a bit strange, but I think it is a side effect of using Antlr's visitor pattern.